packages/playground/blueprints/src/lib/compile.ts

```diff
@@ -48,17 +48,35 @@ export interface CompiledBlueprint {
 	run: (playground: UniversalPHP) => Promise<void>;
 }
 
+export type OnStepProgress = (
+	promise: Promise<void>,
+	step: StepDefinition
+) => void;
 export type OnStepCompleted = (output: any, step: StepDefinition) => any;
+export type OnStepsCompleted = (
+	completedStatus: Map<StepDefinition, boolean>
+) => void;
 
 export interface CompileBlueprintOptions {
 	/** Optional progress tracker to monitor progress */
 	progress?: ProgressTracker;
 	/** Optional semaphore to control access to a shared resource */
 	semaphore?: Semaphore;
 	/** Optional callback with step output */
+	onStepProgress?: OnStepProgress;
 	onStepCompleted?: OnStepCompleted;
+	onStepsCompleted?: OnStepsCompleted;
```

Comment:
The intention seems to be deciding whether to book. How about `shouldBoot` as a name?